sources/packages/backend/apps/api/src/route-controllers/application-exception/_tests_/e2e/application-exception.aest.getPendingApplicationExceptions.e2e-spec.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource } from "typeorm";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import { ApplicationExceptionStatus } from "@sims/sims-db";
+import { getUserFullName } from "../../../../utilities";
+import { TestingModule } from "@nestjs/testing";
+import { saveFakeApplicationWithApplicationException } from "../application-exception-helper";
+
+describe(`ApplicationExceptionAESTController(e2e)-getPendingApplicationExceptions`, () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let appModule: TestingModule;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    appModule = module;
+  });
+
+  it("Should get pending application exceptions when available.", async () => {
+    // Arrange
+    const application1Promise = saveFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Pending,
+      appDataSource,
+      appModule,
+    );
+    const application2Promise = saveFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Pending,
+      appDataSource,
+      appModule,
+    );
+    const application3Promise = saveFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Approved,
+      appDataSource,
+      appModule,
+    );
+    const application4Promise = saveFakeApplicationWithApplicationException(
+      ApplicationExceptionStatus.Declined,
+      appDataSource,
+      appModule,
+    );
+    const [application1, application2, application3, application4] =
+      await Promise.all([
+        application1Promise,
+        application2Promise,
+        application3Promise,
+        application4Promise,
+      ]);
+    const endpoint =
+      "/aest/application-exception?page=0&pageLimit=100&sortField=submittedDate&sortOrder=ASC";
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const applicationExceptionList = response.body.results;
```

Comment:
IMO, all the expected results can be expressed with the object syntax. If I am not wrong it covers "contains" and "do not contains" at the same time and it is easier to follow (IMO).

Reply:
I got the point after our conversation. Since it is the Ministry user, other records can be present that are not related to this test. Nice way to test it.